bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,33 +116,47 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+	public void handleLabels(Control c) {
+	    if (labelDragHandlersAdded) {
+	        return;
+	    }
+	    if (c instanceof Label) {
+	    	long labelViewClass = OS.object_getClass(c.view.id);
+	        addDragHandlers(labelViewClass);// adding the handlers to label class
+	        // If the content view can be an image view, then add the dragging methods to the image view too.
+		    // This is used by Label so that dragging can work even when the Label has an image set on it.
+	        long imageView = 0;
+	        if (((imageView = OS.objc_msgSend(c.view.id, OS.sel_getImageView)) != 0)) {
+	           long cls = OS.object_getClass(imageView);
+	            addDragHandlers(cls);
+	            labelDragHandlersAdded = true;
+	        }
+	    } else if (c instanceof Composite) {
+	        Control[] cal = ((Composite) c).getChildren();
+	        for (Control child : cal) {
+	            handleLabels(child);
+	        }
+	    }
+	}
 void addDragHandlers() {
 	// Our strategy here is to dynamically add methods to the control's class that are required
 	// by NSDraggingDestination. Then, when setTransfer is called, we just register
 	// the types with the Control's NSView and AppKit will call the methods in the protocol
 	// when a drag goes over the view.
-
 	long cls = OS.object_getClass(control.view.id);
+	addDragHandlers(cls);
```

Comment:
Why is addDragHandlers required here?

Reply:
@lshanmug thought both the cls values to be different but its not. I have removed it.